lib/stack_master/aws_driver/s3.rb

```diff
@@ -0,0 +1,70 @@
+require 'digest/md5'
+
+module StackMaster
+  module AwsDriver
+    class S3ConfigurationError < StandardError; end
+
+    class S3
+      attr_accessor :region, :bucket, :prefix, :files
+
+      def initialize(region: nil, bucket: nil, prefix: nil, files: nil)
```

Comment:
@stevehodgkiss I decided that since the S3 region / bucket / prefix are always the same per run, I decided to use the intiialize for the s3 driver to set them, and then use accessors in the rest of the function to get that info. I'm having some trouble writing the rspec test though, since I'm... well... a noob at unit testing. I'm trying to figure it out, but I'm wondering if you know how I should modify those tests? 

They have a number of different "options", which used to be passed specifically to upload_files, but I would prefer to pass to the initializer for the s3 driver. How should I use those different tests as part of different s3_driver#initialize instead of as options passed to the method?

````ruby
  subject(:s3_driver) { StackMaster::AwsDriver::S3.new }
     describe '#upload_files' do
````